### PR TITLE
Send available entities to player on session start

### PR DESCRIPTION
**Pull request by NeutronicMC**

### CODE REVIEW

File: server/session/session.go [hunk 1]
```diff
@@ -406,3 +415,19 @@ func (s *Session) closePlayerList() {
...
 	sessionMu.Unlock()
 }
+
+// sendAvailableEntities sends all registered entities to the player.
+func (s *Session) sendAvailableEntities() {
+	entities := world.Entities()
+	var entityData []actorIdentifier
+	for _, entity := range entities {
+		id := entity.EncodeEntity()
+		s.log.Debugf("entity %v\n", id)
```
Comment (TwistedAsylumMC) [inline server/session/session.go:425]: Guessing this isn't supposed to be here

Comment (NeutronicMC) [reply server/session/session.go:425]: Yep, lemme fix

File: server/world/entity.go [hunk 2]
```diff
@@ -67,3 +67,8 @@ func EntityByName(name string) (SaveableEntity, bool) {
 	e, ok := entities[name]
 	return e, ok
 }
+
+// Entities returns all registered entities.
+func Entities() map[string]SaveableEntity {
```
Comment (Sandertv) [inline server/world/entity.go:72]: I'd make this grab the values from the map and return a slice. We don't want people to be able to modify this map directly.

File: server/session/session.go [hunk 3]
```diff
@@ -104,6 +105,12 @@ type Conn interface {
 	StartGame(data minecraft.GameData) error
 }
 
+// actorIdentifier represents the structure of an actor identifier sent over the network.
+type actorIdentifier struct {
```
Comment (Sandertv) [inline server/session/session.go:109]: I'd also move this down to where it's used. You could even have it inside of the sendAvailableEntities method.

**Review by Sandertv — APPROVED** [review]
Looks good and has been tested.

**Comment by Sandertv** [comment]
Thanks for the PR!

**Pull request merged**